server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryParseRequest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportRequest;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: request to extensions to parse context
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryParseRequest extends TransportRequest {
+
+    private final Class categoryClass;
```

Comment:
I've removed the generic casting on the SDK end, as per your comment on the corresponding SDK PR. Since we already ensure that all category class entries sent over by the extension utilize the NamedWriteable framework, these restrictions are not necessary.